pubsub/gossipsub/gossipsub-v1.2.md

```diff
@@ -0,0 +1,102 @@
+# Gossipsub v1.2
+
+# Overview
+
+This document aims to provide a minimal extension to the [gossipsub
+v1.1](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md)
+protocol.
+
+The proposed extensions are backwards-compatible and aim to enhance the
+efficiency (minimize amplification/duplicates and decrease message latency) of
+the gossip mesh networks for larger messages. 
+
+In more specific terms, a new control message is introduced: `DONTSEND`. It's primarily 
+intended to notify mesh peers that the node already received a message and there is no 
+need to send its duplicate.
+
+# Specification
+
+## Protocol Id
+
+Nodes that support this Gossipsub extension should additionally advertise the
+version number `1.2.0`. Gossipsub nodes can advertise their own protocol-id
+prefix, by default this is `meshsub` giving the default protocol id:
+- `/meshsub/1.2.0`
```

Comment:
Is this necessary, since this is backward compatible? Could just send the the DONTSEND optimistically, and if the peer doesn't handle them, he will just skip them

Reply:
I think we should bump minor when we add a new control msg.

But see my comments in the pr.

Reply:
I think @vyzo and @AgeManning had a similar conversation somewhere regarding protocol versioning. May be they have an opinion on that

Reply:
Bumping the version allows us to filter peers who wont understand the message, so we can save a bit of bandwidth.